lib/thrifter/instrumented_pool.rb

```diff
@@ -0,0 +1,35 @@
+module Thrifter
+  class InstrumentedPool < ConnectionPool
+    attr_reader :statsd
+
+    def initialize(options = { }, &block)
+      super(options, &block)
+      @statsd = options.fetch(:statsd)
+    end
+
+    def checkout(*args)
+      statsd.time('thread_pool.latency') do
+        super.tap do |conn|
+          statsd.increment('thread_pool.checkout')
+          statsd.gauge('thread_pool.in_use', in_use)
+        end
+      end
+    rescue Timeout::Error => ex
```

Comment:
@ahawkins I'm not sure if that is possible but Is it possible to have a test where if raise a timeout error and assert the same?